src/error.rs

```diff
@@ -17,12 +18,28 @@ pub enum ConnectionError {
     Checkout(#[from] diesel::r2d2::Error),
 
     #[error("Failed to issue a query: {0}")]
-    Query(#[from] diesel::result::Error),
+    Query(#[from] DieselError),
 }
 
 /// Syntactic sugar around a Result returning an [`PoolError`].
 pub type PoolResult<R> = Result<R, PoolError>;
 
+/// Async varient of
+/// <https://docs.diesel.rs/master/diesel/prelude/trait.OptionalExtension.html>.
```

Comment:
out of curiosity, can you not use "[`diesel::prelude::OptionalExtension`]" here?

Reply:
Ah, done.